hca/define_api.py

```diff
@@ -1,37 +1,172 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import json, sys, os
-from io import open
+import json
+import os
+import sys
 import requests
-from .parser import get_parser
+
+from io import open
+
+from .parser import get_parser, ADDED_COMMANDS
+from .constants import Constants
 
 
 class API:
     """Class for interacting with the REST Api."""
 
-    def __init__(self, api_url, user):
+    def __init__(self, test=False):
         """Initialize the CLI API."""
-        self.spec = self.get_spec()
-        self.base_url = self.spec['host'] + self.spec['basePath']
-        self.parser = get_parser(self.spec)
+        spec = self.get_spec(test)
+        scheme = spec["schemes"][0] if "schemes" in spec else "https"
+        self.base_url = "{}://{}{}".format(scheme, spec['host'], spec['basePath'])
```

Comment:
consider https://docs.python.org/2/library/urlparse.html, though that might be an overkill.